contracts/package.json

```diff
@@ -65,10 +66,14 @@
     "postinstall": "yarn typechain",
     "precommit": "lint-staged",
     "prettier": "prettier --write \"**/*.{js,json,md,sol,ts}\"",
-    "test": "hardhat test",
+    "test:mainnet": "yarn set-fork:mainnet && hardhat test test/*.test.ts",
+    "test:polygon": "yarn set-fork:polygon && hardhat test test/polygon/*.test.ts; yarn set-fork:mainnet",
+    "test": "yarn test:polygon && yarn test:mainnet",
     "typechain": "hardhat typechain",
     "app:node": "ALCHEMY_API_KEY=$(grep ALCHEMY_API_KEY .env | cut -d '=' -f2) && yarn hardhat node --fork \"https://eth-mainnet.alchemyapi.io/v2/$ALCHEMY_API_KEY\"",
-    "app:setup": "hardhat run scripts/app.ts --network localhost"
+    "app:setup": "hardhat run scripts/app.ts --network localhost",
+    "set-fork:polygon": "sed -i -e 's/eth-mainnet.alchemyapi.io/polygon-mainnet.g.alchemy.com/g' ./hardhat.config.ts && rm -f ./hardhat.config.ts-e",
+    "set-fork:mainnet": "sed -i -e 's/polygon-mainnet.g.alchemy.com/eth-mainnet.alchemyapi.io/g' ./hardhat.config.ts && rm -f ./hardhat.config.ts-e"
```

Comment:
I had an idea for an alternative way of managing this, but wasn't sure it would work, so I coded it up to see: https://github.com/dcgtc/dgrants/commit/256728654705270f775090401b71e6746cdc5d03

This approach uses ENV vars which are defined inline with the package.json commands to drive the fork URL. It feels cleaner to me. What do you think? Feel free to cherry-pick the commit onto this branch if you like the approach.

Reply:
You could also define alternative blockheights in the same if clause if you wanted, though use the same one seems to work fine.

Reply:
There's no native `sed` on Windows so the env var approach is definitely better for cross-platform support. Cherry-picked your commit and modified/rebased it in 98d8330d811363cadd174ac5a26a5c7018bddc11 to set block number also, since I think using a more recent block is probably ideal. 

FWIW, recently had to use the `sed` trick that we do here to adjust a hardcoded bytecode hash in a solidity contract, which is why I defaulted to it without thinking much about it 😅